<a href="https://colab.research.google.com/github/jasonhouse1/TKU-AI-411631327/blob/main/0529.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
n = int(input("请输入整数n: "))
for i in range(1, n+1):
    print('*' * i)


请输入整数n: 5
*
**
***
****
*****


In [ ]:
def determine_season(month):
    if 3 <= month <= 5:
        return "Spring"
    elif 6 <= month <= 8:
        return "Summer"
    elif 9 <= month <= 11:
        return "Autumn"
    elif month == 12 or month == 1 or month == 2:
        return "Winter"
    else:
        return "Input error"

user_input = int(input("Enter a month (1-12): "))
print(f"{user_input} is in {determine_season(user_input)}.")
#多向

Enter a month (1-12): 88
88 is in Input error.


In [8]:
# 季節判斷(巢狀判斷式)
monthB = int(input('請輸入月份(阿拉伯數字): '))

if 1 <= monthB <= 12:
    if 3 <= monthB <= 5:
        print('{}月是春天!'.format(monthB))
    elif 6 <= monthB <= 8:
        print('{}月是夏天!'.format(monthB))
    elif 9 <= monthB <= 11:
        print('{}月是秋天!'.format(monthB))
    else:
        print('{}月是冬天!'.format(monthB))
else:
    print('月份不在範圍內!')


請輸入月份(阿拉伯數字): 5
5月是春天!


In [ ]:
def grade(score):
    if score > 100 or score < 0:
        return "X is error"
    elif 90 <= score <= 100:
        return "X is A"
    elif 80 <= score <= 89:
        return "X is B"
    elif 70 <= score <= 79:
        return "X is C"
    elif 60 <= score <= 69:
        return "X is D"
    elif 0 <= score <= 59:
        return "X is E"
    else:
        return "Input error"
user_score = int(input("请输入成绩（整数型态）: "))

print(grade(user_score))


请输入成绩（整数型态）: 99
X is A


In [5]:
def get_tax_rate(income):
    if income > 2000000:
        return 0.3
    elif income > 1000000:
        return 0.21
    elif income > 600000:
        return 0.13
    elif income > 300000:
        return 0.06
    else:
        return 0  # 若收入在300000以下，稅率為0

def calculate_tax(income, tax_rate):
    return income * tax_rate

# 用戶輸入收入
income = float(input("請輸入今年收入淨額: "))

# 獲取適用的稅率
tax_rate = get_tax_rate(income)

# 計算稅金
tax = calculate_tax(income, tax_rate)

# 輸出結果
print(f"請輸入今年收入淨額: {income:.0f}")
print(f"稅率: {tax_rate:.2f}")
print(f"付稅金額: {tax:.0f}元")


請輸入今年收入淨額: 300001
請輸入今年收入淨額: 300001
稅率: 0.06
付稅金額: 18000元


In [6]:
# 稅金計算(巢狀判斷式)
incomeB = int(input('請輸入今年收入淨額: '))

if incomeB >= 300000:
    if incomeB >= 2000000:
        print('付稅金額: {} 元'.format(incomeB * 0.3))
    elif incomeB >= 1000000:
        print('付稅金額: {} 元'.format(incomeB * 0.21))
    elif incomeB >= 600000:
        print('付稅金額: {} 元'.format(incomeB * 0.13))
    else:
        print('付稅金額: {} 元'.format(incomeB * 0.06))
else:
    print('付稅金額: 0 元')


請輸入今年收入淨額: 1000000
付稅金額: 210000.0 元


In [7]:
# 稅金計算(巢狀判斷式)
incomeB = int(input('請輸入今年收入淨額: '))

if incomeB >= 300000:
    if incomeB >= 2000000:
        tax_rate = 0.3
    elif incomeB >= 1000000:
        tax_rate = 0.21
    elif incomeB >= 600000:
        tax_rate = 0.13
    else:
        tax_rate = 0.06
    tax_amount = incomeB * tax_rate
    print(f'稅率: {tax_rate}')
    print('付稅金額: {} 元'.format(tax_amount))
else:
    print('稅率: 0')
    print('付稅金額: 0 元')


請輸入今年收入淨額: 1000000
稅率: 0.21
付稅金額: 210000.0 元


In [10]:
# 百貨公司週年慶(巢狀判斷式)
shop = int(input('請輸入購物金額: '))

if shop >= 10000:
    if shop >= 100000:
        print('八折 ({}元)'.format(shop * 0.8))
    elif shop >= 50000:
        print('八五折 ({}元)'.format(shop * 0.85))
    elif shop >= 30000:
        print('九折 ({}元)'.format(shop * 0.9))
    else:
        print('九五折 ({}元)'.format(shop * 0.95))
else:
    print('沒打折 {}元'.format(shop))


請輸入購物金額: 9999
沒打折 9999元


In [11]:
#體溫判斷
temperature = float(input("請輸入您的體溫: "))

if temperature >= 37.5:
    print("{:.1f} is over".format(temperature))
elif 37 <= temperature <= 37.4:
    print("{:.1f} is little over".format(temperature))
elif 34 <= temperature <= 36.9:
    print("{:.1f} is normal".format(temperature))
elif temperature < 34:
    print("{:.1f} is error".format(temperature))


請輸入您的體溫: 33
33.0 is error


In [12]:
#成績等第判斷
score = int(input())

if 90 <= score <= 100:
    print('{} is A'.format(score))
elif 80 <= score <= 89:
    print('{} is B'.format(score))
elif 70 <= score <= 79:
    print('{} is C'.format(score))
elif 60 <= score <= 69:
    print('{} is D'.format(score))
elif 0 <= score <= 59:
    print('{} is E'.format(score))
else:
    print('{} is out of range'.format(score))


60
60 is D


In [13]:
#季節判斷(國歷)
month = int(input())

if 3 <= month <= 5:
    print('{} is spring'.format(month))
elif 6 <= month <= 8:
    print('{} is summer'.format(month))
elif 9 <= month <= 11:
    print('{} is autumn'.format(month))
elif month in [12, 1, 2]:
    print('{} is winter'.format(month))
else:
    print('input error')


5
5 is spring


In [14]:
#請使用者輸入一個正整數後,程式就會印出從1到該整數
#(需包括該整數)的數列。(利用range函式)
n = int(input("請輸入一個正整數: "))

for i in range(1, n + 1):
    print(i, end=' ')


請輸入一個正整數: 5
1 2 3 4 5 

In [16]:
#請使用者輸入一個正整數後,程式就會計算出從1
#到該整數的總和。(利用range函式)
sum = 0
n = int(input("請輸入一個正整數: "))

for i in range(1, n + 1):
    sum += i

print("1到%d 的整數和為: %d" % (n, sum))


請輸入一個正整數: 50
1到50 的整數和為: 1275


In [17]:
#FOR迴圈裡面加FOR回圈示範
n = 0

for i in range(1, 10001):
    for j in range(1, 10001):
        n += 1

print(n)


100000000


In [18]:
# 九九乘法表
for i in range(1, 10):
    for j in range(1, 10):
        ans = i * j
        print("%d*%d=%2d" % (i, j, ans), end=" ")
    print()


1*1= 1 1*2= 2 1*3= 3 1*4= 4 1*5= 5 1*6= 6 1*7= 7 1*8= 8 1*9= 9 
2*1= 2 2*2= 4 2*3= 6 2*4= 8 2*5=10 2*6=12 2*7=14 2*8=16 2*9=18 
3*1= 3 3*2= 6 3*3= 9 3*4=12 3*5=15 3*6=18 3*7=21 3*8=24 3*9=27 
4*1= 4 4*2= 8 4*3=12 4*4=16 4*5=20 4*6=24 4*7=28 4*8=32 4*9=36 
5*1= 5 5*2=10 5*3=15 5*4=20 5*5=25 5*6=30 5*7=35 5*8=40 5*9=45 
6*1= 6 6*2=12 6*3=18 6*4=24 6*5=30 6*6=36 6*7=42 6*8=48 6*9=54 
7*1= 7 7*2=14 7*3=21 7*4=28 7*5=35 7*6=42 7*7=49 7*8=56 7*9=63 
8*1= 8 8*2=16 8*3=24 8*4=32 8*5=40 8*6=48 8*7=56 8*8=64 8*9=72 
9*1= 9 9*2=18 9*3=27 9*4=36 9*5=45 9*6=54 9*7=63 9*8=72 9*9=81 


In [ ]:
#1到100的總和
sum = 0
for i in range(1, 101):
    sum += i
print(sum)


In [19]:
#請設計一個程式,讓使用者輸入整數n後,會列出n所有的因數。
#提示:可用for迴圈完成。
n = int(input())
for i in range(1, n + 1):
  if(n%i==0):
    print(i, end=' ')

30
1 2 3 5 6 10 15 30 

In [20]:
#3和7的倍數總和
n = int(input())
sum = 0
for i in range(1,n+1):
  if(i%3==0 or i%7==0):
    sum += i
print(sum)

25
129


In [22]:
#請設計一個程式,將使用者輸入的字串,連續轉換大寫.小寫並print出結果。
#奇數位置為大寫;偶數位置為小寫,如底下範例。

a = input()
for i in range(len(a)):
  if(i%2==0):
    print(a[i].upper(), end='')
  else:
    print(a[i].lower(), end='')

panda
PaNdA

In [24]:
#製作密碼遮罩器
#請設計一個程式,讓使用者input密碼後,再將密碼每個字元轉成「*」後換行輸出,最後也需要換行輸出密碼的長度。
a = input()
for i in range(len(a)):
  print("*",end="")
print()
print("total:{}".format(len(a)))

abcd
****
total:4


In [25]:
#印出99乘法表
n = int(input())
for i in range(n,10):
  for j in range(1,10):
    print("%d*%d=%d"%(i,j,i*j),end=" ")
  print()

5
5*1=5 5*2=10 5*3=15 5*4=20 5*5=25 5*6=30 5*7=35 5*8=40 5*9=45 
6*1=6 6*2=12 6*3=18 6*4=24 6*5=30 6*6=36 6*7=42 6*8=48 6*9=54 
7*1=7 7*2=14 7*3=21 7*4=28 7*5=35 7*6=42 7*7=49 7*8=56 7*9=63 
8*1=8 8*2=16 8*3=24 8*4=32 8*5=40 8*6=48 8*7=56 8*8=64 8*9=72 
9*1=9 9*2=18 9*3=27 9*4=36 9*5=45 9*6=54 9*7=63 9*8=72 9*9=81 


In [26]:
n = int(input())
sum = 0
for i in range(15,n+1):
  if(i%2==0):
    sum += i
print(sum)

22
76


In [ ]:
#成績計算機

# 讀取並分割用戶輸入的成績
scores = input().split()

# 將成績轉換為浮點數
chinese = float(scores[0])
english = float(scores[1])
math = float(scores[2])

# 計算總分與平均分數
total_score = chinese + english + math
average_score = total_score / 3

# 輸出總分與平均分數（四捨五入到小數後2位）
print("sum={:.2f}".format(round(total_score, 2)))
print("avg={:.2f}".format(round(average_score, 2)))


In [ ]:
#BMI計算(分兩行輸入)

# 讀取身高與體重並轉換為浮點數
height = float(input())
weight = float(input())

# 計算BMI
bmi = weight / ((height / 100) ** 2)

# 以特定格式輸出BMI結果，輸出至小數後2位
print("%.2f" % bmi)


print文字
Description

使用print功能顯示文字



Input
Hello World


Output
Hello World


Sample Input 1

Hello World
Sample Output 1

Hello World

In [27]:
num = int(input())

if num >= 0:
    if num % 2 == 0:
        print("{} is even".format(num))
    else:
        print("{} is odd".format(num))
else:
    print("Input Error")


0
0 is even


print文字
Input
Hello World


Output
Hello World


Sample Input 1

Hello World
Sample Output 1

Hello World


In [28]:
text = input()
print(text)


Hello World
Hello World


取得字串長度

In [ ]:
def get_length(string):
    return len(string)

input_string = input()
print(get_length(input_string))


四則運算-加法
Description

加法運算


Input
456+789


Output
1245


Sample Input 1

456+789
Sample Output 1

1245

In [29]:
expression = input()
result = eval(expression)
print(result)


456+789
1245


複列印文字
Description

將1,2,3在括弧內印出3次


Input
在同一行輸入1 2 3


Output
重複打印結果


Sample Input 1

1 2 3
Sample Output 1

1 2 3 1 2 3 1 2 3

In [30]:
numbers = input().split()
repeated_numbers = ' '.join(numbers * 3)
print(repeated_numbers)


1 2 3
1 2 3 1 2 3 1 2 3


將文字順序顛倒印出
Description

反向輸出列表中的數值


Input
在同一行輸入3個單字


Output
文字將顛倒印出


Sample Input 1

apple orange banana
Sample Output 1

banana orange apple

In [31]:
words = input().split()
reversed_words = ' '.join(words[::-1])
print(reversed_words)


apple orange banana
banana orange apple


求解1到100的總和
Description

1+2+3+..............+99+100=?


Input
整數1到100


Output
總和


Sample Input 1

1+2+3+4+5+.........+99+100
Sample Output 1

5050

In [ ]:
# 初始化總和變數
total = 0

# 使用迴圈計算總和
for i in range(1, 101):
    total += i

# 輸出結果
print(total)


季節判斷
Description

我們知道1～3月為春天(spring)；4～6月為夏天(summer)；7～9月為秋天(autumn)；10～12月為冬天(winter)。

請設計程式，讓使用者輸入月份(阿拉伯數字)後，請輸出該月份為哪個季節。若輸入非上述月份，請輸出錯誤結果。


Input
(none)


Output
(請參考下方Sample輸出結果)

春天為 spring

夏天為 summer

秋天為 autumn

冬天為 winter

輸入錯誤為 Input Error


Sample Input 1

2
Sample Output 1

2 is spring
Sample Input 2

6
Sample Output 2

6 is summer
Sample Input 3

9
Sample Output 3

9 is autumn
Sample Input 4

11
Sample Output 4

11 is winter
Sample Input 5

14
Sample Output 5

Input Error
Sample Input 6

-2
Sample Output 6

Input Error

In [32]:
month = int(input())

if month in range(1, 4):
    season = "spring"
elif month in range(4, 7):
    season = "summer"
elif month in range(7, 10):
    season = "autumn"
elif month in range(10, 13):
    season = "winter"
else:
    season = "Input Error"

print("{} is {}".format(month, season))


5
5 is summer


計程車費用問題
Description

狗仔姜賢南為了要蒐集證據而跟蹤他人，搭上了一台計程車。計程車計費方式為前5公里共240元，5公里後，只要多1公里加收40元(意思是第6公里起加收40元)，請設計程式讓司機能夠輸入公里數(int型態)，來計算乘車費用。


Input
(none)


Output
(請參考下方Sample輸出結果)

如果公里數為0，則輸出 0 yuan

如果公里數為1~5之間，則皆輸出 240 yuan

如果公里數大於5，則自行計算總價格為多少。 例如公里數為8，則輸出360 yuan

如果公里數為負數，則輸出Input Error


Sample Input 1

-2
Sample Output 1

Input Error
Sample Input 2

0
Sample Output 2

0 yuan
Sample Input 3

1
Sample Output 3

240 yuan
Sample Input 4

5
Sample Output 4

240 yuan
Sample Input 5

6
Sample Output 5

280 yuan
Sample Input 6

10
Sample Output 6

440 yuan

In [33]:
distance = int(input())

if distance < 0:
    print("Input Error")
elif distance == 0:
    print("0 yuan")
elif distance <= 5:
    print("240 yuan")
else:
    total = 240 + (distance - 5) * 40
    print("{} yuan".format(total))


5
240 yuan


銀行複利問題
Description

同珢在銀行有一筆存款，他的本金為10,0000，目前銀行存款年利率為3%，以複利計算，N年後本金為何？

請讓使用者自行出數N年數值(為int型態)。

複利公式為：本金 ×（1 ＋ 利率） N的次方

==================================

舉例：本金為1000元，銀行年利率為1%，以複利計算，6年後本金為何？

解法：1000 x (1 + 0.01)  6次方


Input
(none)


Output
(請參考下方Sample輸出結果)

如果輸入的年份小於0，則輸出Input Error

如果輸入的年份大於等於0，則輸出計算的結果


Sample Input 1

0
Sample Output 1

100000.0
Sample Input 2

1
Sample Output 2

103000.0
Sample Input 3

2
Sample Output 3

106090.0
Sample Input 4

5
Sample Output 4

115927.40743
Sample Input 5

12
Sample Output 5

142576.08868461792
Sample Input 6

-2
Sample Output 6

Input Error

In [34]:
principal = 100000
interest_rate = 0.03

years = int(input())

if years < 0:
    print("Input Error")
else:
    amount = principal * (1 + interest_rate) ** years
    print("{:.5f}".format(amount))


12
142576.08868


找出3個數的最小值
Description

請設計一個程式，讓使用者輸入3個整數(用空格隔開)後，來找出3個數的最小值。

提示：可用判斷式完成。



假設使用者input(用空格隔開)： 99 50 80

程式要output：50 (因為50為3數之間最小值)


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

10 30 20
Sample Output 1

10
Sample Input 2

5 8 7
Sample Output 2

5
Sample Input 3

0 8 7
Sample Output 3

0
Sample Input 4

-10 3 -20
Sample Output 4

-20

In [35]:
numbers = input().split()

# 將輸入轉換為整數
numbers = [int(num) for num in numbers]

# 找出最小值並輸出
print(min(numbers))


0 8 7
0


找出3個數的最大值
Description

請設計一個程式，讓使用者輸入3個浮點數(用空格隔開)後，來找出3個數的最大值。

提示：可用判斷式完成。



假設使用者input(用空格隔開)： 99.5 50.9 80.3

程式要output：99.5 (因為99.5為3數之間最大值)


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

10.3 30.9 20.8
Sample Output 1

30.9
Sample Input 2

5.0 8.7 7.8
Sample Output 2

8.7
Sample Input 3

0 8.1 7.9
Sample Output 3

8.1
Sample Input 4

-10.3 3.999 -20.8
Sample Output 4

3.999

In [36]:
numbers = input().split()

# 將輸入轉換為浮點數
numbers = [float(num) for num in numbers]

# 找出最大值並輸出
print(max(numbers))


-10.3 3.999 -20.8
3.999


列出所有因數
Description

請設計一個程式，讓使用者輸入整數 n 後，會列出 n 所有的因數。

提示：可用for迴圈完成。



假設使用者input： 30

程式要output：1 2 3 5 6 10 15 30 (也就是30的所有因數)


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

17
Sample Output 1

1 17
Sample Input 2

35
Sample Output 2

1 5 7 35
Sample Input 3

87
Sample Output 3

1 3 29 87
Sample Input 4

520
Sample Output 4

1 2 4 5 8 10 13 20 26 40 52 65 104 130 260 520

In [37]:
n = int(input())
factors = []

for i in range(1, n + 1):
    if n % i == 0:
        factors.append(i)

# 輸出所有因數
print(*factors)


17
1 17


計算3或7的倍數總和
Description

請設計一個程式，讓使用者輸入整數 n 後，會計算 1 ~ n中，3或7的倍數總和。

提示：可用for迴圈 + if判斷式 完成。



假設使用者input： 25

程式要output：129

(因為1~25中，3的倍數有：3 6 9 12 15 18 21 24)

(因為1~25中，7的倍數有：7 14 21)

故 3+6+7+9+12+14+15+18+21+24 = 129


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

20
Sample Output 1

84
Sample Input 2

50
Sample Output 2

541
Sample Input 3

100
Sample Output 3

2208
Sample Input 4

9999
Sample Output 4

21426429

In [38]:
n = int(input())
total = 0

for i in range(1, n + 1):
    if i % 3 == 0 or i % 7 == 0:
        total += i

print(total)


20
84


用＊印出靠左下的直角三角形
Description

請設計一個程式，讓使用者輸入整數 n 後，會用＊印出一個靠左下的直角三角形。



假設使用者input： 4

程式要output：擷取.PNG

output的＊與＊之間不需要空格！


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

5
Sample Output 1

*
**
***
****
*****
Sample Input 2

8
Sample Output 2

*
**
***
****
*****
******
*******
********
Sample Input 3

3
Sample Output 3

*
**
***
Sample Input 4

10
Sample Output 4

*
**
***
****
*****
******
*******
********
*********
**********

In [39]:
n = int(input())

for i in range(1, n + 1):
    for j in range(i):
        print("*", end="")
    print()



4
*
**
***
****


用連續數字印出靠左下的直角三角形
Description

請設計一個程式，讓使用者輸入整數 n 後，會用連續數字印出一個靠左下的直角三角形。

假設使用者input： 4

程式要output：擷取.PNG

output的數字與數字之間不需要空格！


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

3
Sample Output 1

1
12
123
Sample Input 2

5
Sample Output 2

1
12
123
1234
12345
Sample Input 3

7
Sample Output 3

1
12
123
1234
12345
123456
1234567
Sample Input 4

13
Sample Output 4

1
12
123
1234
12345
123456
1234567
12345678
123456789
12345678910
1234567891011
123456789101112
12345678910111213

In [40]:
n = int(input())

for i in range(1, n + 1):
    for j in range(1, i + 1):
        print(j, end="")
    print()


13
1
12
123
1234
12345
123456
1234567
12345678
123456789
12345678910
1234567891011
123456789101112
12345678910111213


求兩數的最大公因數
Description

請設計一個程式，讓使用者在同一行輸入2個整數後，計算並輸出兩數的最大公因數。



假設使用者input： 10 15

程式要output：5


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

60 20
Sample Output 1

20
Sample Input 2

80 81
Sample Output 2

1
Sample Input 3

90 99
Sample Output 3

9
Sample Input 4

120 150
Sample Output 4

30

In [41]:
def gcd(a, b):
    while b != 0:
        a, b = b, a % b
    return a

# 讀取兩個整數
a, b = map(int, input().split())

# 計算最大公因數並輸出結果
print(gcd(a, b))


80 81
1


求兩數的最小公倍數
Description

請設計一個程式，讓使用者在同一行輸入2個整數後，計算並輸出兩數的最小公倍數。



假設使用者input： 10 15

程式要output：30


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

20 50
Sample Output 1

100
Sample Input 2

11 13
Sample Output 2

143
Sample Input 3

98 99
Sample Output 3

9702
Sample Input 4

113 200
Sample Output 4

22600

In [42]:
def gcd(a, b):
    while b != 0:
        a, b = b, a % b
    return a

def lcm(a, b):
    return (a * b) // gcd(a, b)

# 讀取兩個整數
a, b = map(int, input().split())

# 計算最小公倍數並輸出結果
print(lcm(a, b))


113 200
22600


判斷是否構成三角形
Description

請設計一個程式，讓使用者在同一行input 3個數(整數int型態)，並判斷是否構成三角形。



構成三角形要素：任兩邊加起來要大於第三邊。



若構成，請輸出Valid

若不構成，請輸出InValid


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

2 5 6
Sample Output 1

Valid
Sample Input 2

2 5 7
Sample Output 2

Invalid
Sample Input 3

28 29 31
Sample Output 3

Valid
Sample Input 4

2 4 7
Sample Output 4

Invalid

In [43]:
def is_triangle(a, b, c):
    if a + b > c and a + c > b and b + c > a:
        return "Valid"
    else:
        return "Invalid"

# 讀取三個整數
a, b, c = map(int, input().split())

# 判斷是否構成三角形並輸出結果
print(is_triangle(a, b, c))


28 29 31
Valid


大小寫變變樂

In [44]:
def change_case(s):
    result = ''
    for i in range(len(s)):
        if i % 2 == 0:  # 如果索引是偶數，則轉換為大寫
            result += s[i].upper()
        else:  # 如果索引是奇數，則轉換為小寫
            result += s[i].lower()
    return result

# 讀取使用者輸入的字串
s = input()

# 呼叫函式進行大小寫轉換並輸出結果
print(change_case(s))


panda
PaNdA


將字串(list)每一項倒序印出來
Description

請設計一個程式，讓使用者在同一行input 資料至字串(list)裡面，再將字串(list)每一項倒序印出來，並output在同一行。


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

apple banana cake
Sample Output 1

cake banana apple
Sample Input 2

30 60 90 120 150
Sample Output 2

150 120 90 60 30
Sample Input 3

panda is cute and lovely
Sample Output 3

lovely and cute is panda
Sample Input 4

one people emo , two people wemo
Sample Output 4

wemo people two , emo people one

In [45]:
# 讀取使用者輸入的字串
s = input()

# 將字串分割成列表
words = s.split()

# 將列表中的每個字串倒序排列
reversed_words = list(reversed(words))

# 將排列後的列表組合成一個新的字串
result = ' '.join(reversed_words)

# 輸出結果
print(result)


apple banana cake
cake banana apple


時間秒數轉換
Description

我們知道一天有24小時(共86400秒)。

請設計一個程式，讓使用者input 時間的秒數後，轉換成格式為24小時制的『時：分：秒』。

請注意：時、分、秒皆需補零(皆2位數)



範例input： 234

範例output：00:03:54

（也就是234秒，換算後是3分鐘54秒，故需輸出00:03:54）


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

3845
Sample Output 1

01:04:05
Sample Input 2

5438
Sample Output 2

01:30:38
Sample Input 3

52087
Sample Output 3

14:28:07
Sample Input 4

59438
Sample Output 4

16:30:38
Sample Input 5

86399
Sample Output 5

23:59:59

In [46]:
# 讀取使用者輸入的秒數
seconds = int(input())

# 計算時、分、秒
hours = seconds // 3600
minutes = (seconds % 3600) // 60
remaining_seconds = seconds % 60

# 格式化時間字串，補零至兩位數
time_str = '{:02d}:{:02d}:{:02d}'.format(hours, minutes, remaining_seconds)

# 輸出結果
print(time_str)


59438
16:30:38


判斷平年、閏年
Description

我們知道閏年的2月有 29 天，平年的2月則是 28 天。

請設計一個程式，讓使用者input 西元年後，輸出該年是平年或閏年。



判斷閏年的規則：

1.除以 4 能整除，且除以 100 不能整除

2. 如果剛好是 100 的倍數，且除以 400 能整除



舉例來說 ：

2000 年是 100 的倍數且除以 400 能整除，所以是 2000 年是閏年。

2100 年雖然是 4 的倍數，但除以 100 能整除，所以 2100 年是平年。


Input
(請參考下方input sample)


Output
(請參考下方output sample)



平年：common year

閏年：leap year

輸出格式為： 『年份 is common year』  或  『年份 is leap year』


Sample Input 1

2000
Sample Output 1

2000 is leap year
Sample Input 2

2020
Sample Output 2

2020 is leap year
Sample Input 3

2097
Sample Output 3

2100 is common year
Sample Input 4

2100
Sample Output 4

2100 is common year

In [47]:
# 讀取使用者輸入的年份
year = int(input())

# 判斷是否為閏年
if (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0):
    print(year, 'is leap year')
else:
    print(year, 'is common year')


2020
2020 is leap year


偶數加總
Description

讓使用者input一個數字後，會輸出15到該數之間(含該數)的所有偶數之總和。



範例輸入：22

範例輸出：76

介於15至22之間的偶數有：16 18 20 22

故16+18+20+22 = 76


Input
(請參考下方input sample)


Output
(請參考下方output sample)


Sample Input 1

20
Sample Output 1

54
Sample Input 2

30
Sample Output 2

184
Sample Input 3

45
Sample Output 3

450
Sample Input 4

99
Sample Output 4

2394

In [48]:
num = int(input())

total = 0
for i in range(15, num + 1):
    if i % 2 == 0:
        total += i

print(total)


20
54


Print字串
Description

請嘗試print出使用者input的字串，並依照output要求而輸出。



注意：系統會自動帶入底下input值至你的程式碼做驗證，故請勿使用暴力解法。


Input
(參考下方input sample)


Output
(參考下方output sample)

輸出結果為Ans + 空格 + 使用者input的字串



範例input： apple

範例output： Ans apple



範例input： cake

範例output： Ans cake


Sample Input 1

Hello World
Sample Output 1

Ans Hello World
Sample Input 2

I like Python
Sample Output 2

Ans I like Python
Sample Input 3

Panda is cute
Sample Output 3

Ans Panda is cute
Sample Input 4

9487
Sample Output 4

Ans 9487
Hint

可用字串相加的方式。

In [49]:
user_input = input()
print("Ans", user_input)


9487
Ans 9487


利用for迴圈產生指定數列
Description

請設計一個程式，讓使用者在同一行同時輸入range三個參數 起始值、終止值、間隔值(整數型態)後，利用 for 迴圈來輸出對應的數列，產出來的數字應透過「,」進行分隔。


Input
讓使用者在同一行依序輸入range三個參數起始值、終止值、間隔值(整數型態)


Output
(請參考sample output)

產出來數列後，每個數字應透過「,」進行分隔。


Sample Input 1

0 100 5
Sample Output 1

0,5,10,15,20,25,30,35,40,45,50,55,60,65,70,75,80,85,90,95,
Sample Input 2

1 200 4
Sample Output 2

1,5,9,13,17,21,25,29,33,37,41,45,49,53,57,61,65,69,73,77,81,85,89,93,97,101,105,109,113,117,121,125,129,133,137,141,145,149,153,157,161,165,169,173,177,181,185,189,193,197,
Sample Input 3

10 500 25
Sample Output 3

10,35,60,85,110,135,160,185,210,235,260,285,310,335,360,385,410,435,460,485,
Sample Input 4

15 19 1
Sample Output 4

15,16,17,18,
Sample Input 5

-10 -1 2
Sample Output 5

-10,-8,-6,-4,-2,

In [50]:
start, end, step = map(int, input().split())
for i in range(start, end + 1, step):
    print(i, end=",")


-10 -1 2
-10,-8,-6,-4,-2,

學號總和
Description

讓使用者輸入學號，並輸出學號各碼的總和結果。



注意：系統會自動帶入底下input值至你的程式碼做驗證，故請勿使用暴力解法。


Input
(參考下方input sample)

系統預設帶入input值為學號(用空格分開每個數字)。


Output
(參考下方output sample)

請將學號拆開的9碼，輸出其加總。


Sample Input 1

4 0 8 6 3 0 8 4 5
Sample Output 1

38
Sample Input 2

6 1 1 6 3 0 3 9 2
Sample Output 2

31
Sample Input 3

6 1 1 6 3 0 4 0 0
Sample Output 3

21
Hint

?  =map(int,input().split())

總共定義9個變數(因為學號有9碼)

In [53]:
digits = map(int, input().split())
total = sum(digits)
print(total)


4 0 8 6 3 0 8 4 5


TypeError: 'int' object is not callable

停車費計算
Description

請設計一個程式，讓使用者輸入停車的分鐘數(整數int型態)，來計算該使用者需要付的停車費用。



★停車場費用計算規則：

停車滿一個小時$40元，如果未滿一個小時則收費20元。

當日停車費最高收費為300元。



範例：

30分鐘 --> 20元

61分鐘 --> 60元 (40+20)

121分鐘 --> 100元 (40+40+20)

上限為300元，超過300元則視為300元。




Input
使用者輸入分鐘數(整數型態)


Output
輸出格式

park time:停車時間(小時):停車時間(分鐘)

pay:停車費用money


Sample Input 1

669
Sample Output 1

park time:11h:9m
pay:300 money
Sample Input 2

569
Sample Output 2

park time:9h:29m
pay:300 money
Sample Input 3

469
Sample Output 3

park time:7h:49m
pay:300 money
Sample Input 4

369
Sample Output 4

park time:6h:9m
pay:260 money
Sample Input 5

269
Sample Output 5

park time:4h:29m
pay:180 money
Sample Input 6

69
Sample Output 6

park time:1h:9m
pay:60 money

In [57]:
# 讀取停車時間（分鐘）
parking_time = int(input())

# 計算小時和分鐘
hours = parking_time // 60
minutes = parking_time % 60

# 計算停車費用
if parking_time <= 60:
    fee = 20
elif parking_time <= 300:
    fee = 20 + (hours - 1) * 40
    if minutes > 0:
        fee += 40
else:
    fee = 300

# 輸出結果
print("park time:{}h:{}m".format(hours, minutes))
print("pay:{} money".format(fee))


269
park time:4h:29m
pay:180 money


將學號總和開根號後，再判斷奇偶數
Description

請設計一個程式，讓使用者在同一行依序輸入學號 9 碼(整數型態)後，並分別輸出總和、開根號結果、開根號強制轉型為整數int型態、強制轉型後的結果為奇數或偶數。



變數名稱命名參考：

總和 sum

開根號結果 squ

開根號強制轉型為整數型態 squi



注意：系統會自動帶入底下input值至你的程式碼做驗證，故請勿使用暴力解法。


Input
(參考下方input sample)

系統預設帶入input值為學號(用空格分開每個數字)。


Output
(參考下方output sample)

分四行分別輸出：學號的總和、開根號結果、開根號強制轉型為整數int型態、強制轉型後的結果為奇數或偶數。


Sample Input 1

6 1 2 6 3 0 0 2 9
Sample Output 1

sum:29
squ:5.385164807134504
squi:5
squi is odd
Sample Input 2

6 1 1 6 3 0 3 9 2
Sample Output 2

sum:31
squ:5.5677643628300215
squi:5
squi is odd
Sample Input 3

6 1 1 6 3 0 4 0 0
Sample Output 3

sum:21
squ:4.58257569495584
squi:4
squi is even

In [56]:
from math import sqrt

# 讀取學號
digits = list(map(int, input().split()))

# 計算總和
total = sum(digits)
print("sum:{}".format(total))

# 計算開根號
squ = sqrt(total)
print("squ:{}".format(squ))

# 強制轉型為整數
squi = int(squ)
print("squi:{}".format(squi))

# 判斷奇偶數
if squi % 2 == 0:
    print("squi is even")
else:
    print("squi is odd")


611630400


TypeError: 'int' object is not callable